code/modules/mining/equipment/kinetic_crusher.dm

```diff
@@ -146,6 +125,41 @@
 					C.total_damage += detonation_damage
 				L.apply_damage(detonation_damage, BRUTE, blocked = def_check)
 
+/obj/item/kinetic_crusher/attack_secondary(atom/target, mob/living/user) //This is for aiming at yourself/adjacent mobs
+	if(target != user && wielded)
+		fire_kinetic_blast(target, user)
+	if(target == user)
+		to_chat(user, span_warning("You can't aim [src] at yourself!"))
+	if(!wielded)
+		to_chat(user, span_warning("[src] is too heavy to use with one hand! You fumble and drop everything."))
+		user.drop_all_held_items()
+	return SECONDARY_ATTACK_CANCEL_ATTACK_CHAIN
```

Comment:
the order in this proc is nonsensical

Reply:
to elaborate:
you shouldnt be checking
if case 1 and case 2 succeed
if case 1 fail
if case 2 fail
this is inefficient, youre doing checks twice and it looks dumb
way better is
if case 1 fail
if case 2 fail
succeed